plugins/mcp-actions-backend/src/services/McpService.ts

```diff
@@ -72,25 +72,41 @@ export class McpService {
         throw new NotFoundError(`Action "${params.name}" not found`);
       }
 
-      const { output } = await this.actions.invoke({
-        id: action.id,
-        input: params.arguments as JsonObject,
-        credentials,
-      });
+      try {
+        const { output } = await this.actions.invoke({
+          id: action.id,
+          input: params.arguments as JsonObject,
+          credentials,
+        });
 
-      return {
-        // todo(blam): unfortunately structuredContent is not supported by most clients yet.
-        // so the validation for the output happens in the default actions registry
-        // and we return it as json text instead for now.
-        content: [
-          {
-            type: 'text',
-            text: ['```json', JSON.stringify(output, null, 2), '```'].join(
-              '\n',
-            ),
-          },
-        ],
-      };
+        return {
+          // todo(blam): unfortunately structuredContent is not supported by most clients yet.
+          // so the validation for the output happens in the default actions registry
+          // and we return it as json text instead for now.
+          content: [
+            {
+              type: 'text',
+              text: ['```json', JSON.stringify(output, null, 2), '```'].join(
+                '\n',
+              ),
+            },
+          ],
+        };
+      } catch (e) {
+        return {
+          content: [
+            {
+              type: 'text',
+              text: [
+                '```json',
+                JSON.stringify(e?.body?.error?.cause ?? e, null, 2),
```

Comment:
I'm feeling this assumes a bit too much about the error coming back from the actions client. Thinking it's best to check if it's actually a `ResponseError` and unpack it like this in that case? Otherwise serialize more directly. We may need to be a bit careful with the serialization as well, since `JSON.stringify` on a regular error instance doesn't work too well. We could use `serializeError` from `@backstage/errors`, and by that also make sure that `includeStack` isn't set.